homework3/EVPurchaseProgram2/src/main/java/ro/siit/java10/evp/CarSorter.java

```diff
@@ -0,0 +1,139 @@
+package ro.siit.java10.evp;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+
+public class CarSorter {
+
+    public static ArrayList<DatabaseElement> sortCarsByPrice(ArrayList<DatabaseElement> initialCarList,
+                                                             boolean inAscendingOrder) {
+
+        Comparator<DatabaseElement> priceComparator;
+        // daca sortarea se face in ordine crescatoare
+
+        if (inAscendingOrder == true) {
```

Comment:
You can refactor this by extracting this in a separate class and based on the ascending/ descending ordering return (-1) * compared value.
Example:

```java

public class DBComparator implements Comparator< DatabaseElement > {

  public enum SortOrder {ASCENDING, DESCENDING}

  private SortOrder sortOrder;

  public DBComparator(SortOrder sortOrder) {
    this.sortOrder = sortOrder;
  }

  @Override
  public int compare(DatabaseElement o1, DatabaseElement o2) {
    Integer price1 = o1.getPrice();
    Integer price2 = o2.getPrice();
    int compare = Math.signum(price1.compareTo(price2));

    if (sortOrder == ASCENDING) {
      return compare;
    } else {
      return compare * (-1);
    }
  }
}
```